bdk_electrum_example/src/main.rs

```diff
@@ -89,137 +122,141 @@ fn main() -> anyhow::Result<()> {
+                    eprintln!("scanning {:?}", index);
+                    script
+                })));
+            }
+            if unused_spks {
+                let unused_spks = tracker
+                    .txout_index
+                    .unused_spks(..)
+                    .map(|(k, v)| (k.clone(), v.clone()))
+                    .collect::<Vec<_>>();
+                spks = Box::new(spks.chain(unused_spks.into_iter().map(|(index, script)| {
+                    eprintln!("Checking if address at {:?} has been used", index);
+                    script
+                })));
+            }
+            if unspent_spks {
+                let unspent_txouts = tracker
+                    .full_utxos()
+                    .map(|(k, v)| (k.clone(), v.clone()))
+                    .collect::<Vec<_>>();
+                spks = Box::new(
+                    spks.chain(unspent_txouts.into_iter().map(|(_index, ftxout)| {
+                        eprintln!("checking if {} has been spent", ftxout.outpoint);
+                        ftxout.txout.script_pubkey
+                    })),
+                );
+            }
+
+            let mut outpoints = Vec::new();
+            if unspent_outpoints {
+                outpoints = tracker.full_utxos().map(|(_, txo)| txo.outpoint).collect();
+            }
 
-            (new_sparsechain, BTreeMap::default())
+            let mut txids = Vec::new();
+            if unconfirmed_txs {
+                txids = tracker
+                    .chain()
+                    .range_txids_by_height(TxHeight::Unconfirmed..)
+                    .map(|(_, txid)| *txid)
+                    .collect();
+            }
+
+            let local_chain = tracker.chain().checkpoints().clone();
+
+            // drop lock on tracker
+            drop(tracker);
+
+            // we scan the spks **without** a lock on the tracker
+            ElectrumUpdate {
+                chain_update: client
+                    .scan_without_keychain(
+                        &local_chain,
+                        spks,
+                        txids,
+                        outpoints,
+                        scan_option.batch_size,
+                    )
+                    .context("scanning the blockchain")?,
+                ..Default::default()
+            }
         }
     };
 
-    let new_txids = {
-        let tracker = &*tracker.lock().unwrap();
-        chain_update
-            .txids()
-            .filter(|(_, txid)| tracker.graph().get_tx(*txid).is_none())
-            .map(|&(_, txid)| txid)
-            .collect::<Vec<_>>()
-    };
+    let missing_txids = response.find_missing_txids(&*tracker.lock().unwrap());
 
     // fetch the missing full transactions **without** a lock on the tracker
     let new_txs = client
-        .batch_transaction_get(new_txids.iter())
+        .batch_transaction_get(missing_txids)
         .context("fetching full transactions")?;
```

Comment:
I'm wondering if it would make sense to have a `fetch_missing_txs` that basically merges `find_missing_txids` and the `batch_transaction_get`, instead of `find_missing_txids`.

I don't think anyone would use `find_missing_txids` if it's not for then fetching them, and having a method that does both operations might be clearer for the user

Reply:
I think we should defer this.